### PR TITLE
task was solved

### PR DESCRIPTION
**Pull request by vanyachyzh**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden === false) {
```
[the remark this answers is not in the document]
Comment (vanyachyzh) [reply src/herbivoresAndCarnivores.js:26]: I use (victim.hidden === false) not only for checking of hidden property, but also to check type of victim. If victim is carnivores victim.hidden will be undefined -> false

[the remark this answers is not in the document]
Comment (vanyachyzh) [reply src/herbivoresAndCarnivores.js:26]: I have to bite another animal if it is a herbivore only

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      const onlyAlive = Animal.alive.filter(animal => {
+        return animal.name !== victim.name;
+      });
+
+      Animal.alive = onlyAlive;
```
Comment (Viktor-Kost) [inline src/herbivoresAndCarnivores.js:35]: this onlyAlive is redundant
```suggestion
      Animal.alive = Animal.alive.filter(animal => {
        return animal.name !== victim.name;
      });
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Nice!

Left you some comments

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      Animal.alive = Animal.alive.filter(animal => {
+        return animal.name !== victim.name;
+      });
```
Comment (OlEzhA0) [inline src/herbivoresAndCarnivores.js:33]:
```suggestion
      Animal.alive = Animal.alive.filter(animal => animal.name !== victim.name);
```

**Review by OlEzhA0 — APPROVED** [review]
hey! just small improvement to your code

**Comment by VitaliyBondarenko1982** [comment]
This pull already approved. You don't need to re-request again )

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**